src/extension/analytics.js

```diff
@@ -0,0 +1,107 @@
+import * as cache from './cache';
+import { getHostEnvName } from '../clients/server-env';
+import ua from 'universal-analytics';
+import uuid from 'uuid/v4';
+import _ from 'lodash';
+import { getLocalDataClient } from '../clients/clients-factory';
+
+function getTrackingId() {
+  return getHostEnvName() === 'production' ? 'UA-807293-5' : 'UA-807293-12';
+}
+
+async function getClientId() {
+  return await cache.get('ga-client-id', null, () => uuid());
+}
+
+async function getUaVisitor() {
+  const clientId = await getClientId();
+
+  const visitor = ua(getTrackingId(), clientId);
+  reportData.clientId = clientId;
+
+  const email = await getLocalDataClient().loadUserEmail();
+  if (email) {
+    visitor.set('userId', email);
+    reportData.userId = email;
+  }
+  visitor.set('isDeveloper', true);
+  reportData.isDeveloper = true;
+
+  return visitor;
+}
+
+async function reportEvent({ category, action, label }) {
+  const visitor = await getUaVisitor();
+
+  await visitor.event(category, action, label).send(err => {
+    if (err) {
+      console.error(err);
+    } else if (getHostEnvName() !== 'production') {
+      console.error('GA Report completed', {
```

Comment:
To avoid interference with spinner which is using stdout.

Outputs such as this should be hidden behind --verbosive flag which should disable all spinners and progress bars to allow for verbosive outputs.